lib/bindings/HTTPBindings.js

```diff
@@ -165,8 +165,25 @@ function handleIncomingMeasure(req, res, next) {
         }
     }
 
-    function extractServiceAndSubservice(group, callback) {
-        callback(null, group.service, group.subservice);
+    function findOrCreate(group, callback) {
+        iotAgentLib.getDevice(req.deviceId, group.service, group.subservice, function(error, device) {
+            if (!error && device) {
+                callback(null, device);
+            } else if (error.name === 'DEVICE_NOT_FOUND') {
+                var newDevice = {
+                    id: req.deviceId,
+                    service: group.service,
+                    subservice: group.subservice,
+                    type: group.type
+                };
+
+                iotAgentLib.register(newDevice, function(error, device) {
```

Comment:
Any reason not to pass `callback` directly as the second argument to `iotAgentLib.register()`?